test/Microsoft.Data.Sqlite.Tests/SqliteConnectionTest.cs

```diff
@@ -387,6 +388,397 @@ public void CreateCollation_works_with_state()
+            {
+                connection.Open();
+                connection.CreateFunction<long>("test", () => throw new SqliteException("Test", 200));
+
+                var ex = Assert.Throws<SqliteException>(() => connection.ExecuteScalar<long>("SELECT test();"));
+
+                Assert.Equal(Resources.SqliteNativeError(200, "Test"), ex.Message);
+                Assert.Equal(200, ex.SqliteErrorCode);
+            }
+        }
+
+        [Fact]
+        public void CreateFunction_works_when_null()
+        {
+            using (var connection = new SqliteConnection("Data Source=:memory:"))
+            {
+                connection.Open();
+                connection.CreateFunction("test", () => 1L);
+                connection.CreateFunction("test", default(Func<long>));
+
+                var ex = Assert.Throws<SqliteException>(() => connection.ExecuteScalar<long>("SELECT test();"));
+
+                Assert.Equal(Resources.SqliteNativeError(raw.SQLITE_ERROR, "no such function: test"), ex.Message);
+                Assert.Equal(raw.SQLITE_ERROR, ex.SqliteErrorCode);
+            }
+        }
+
+        [Fact]
+        public void CreateFunction_works_when_result_null()
+        {
+            using (var connection = new SqliteConnection("Data Source=:memory:"))
+            {
+                connection.Open();
+                connection.CreateFunction<object>("test", () => null);
+
+                var result = connection.ExecuteScalar<object>("SELECT test();");
+
+                Assert.Equal(DBNull.Value, result);
+            }
+        }
+
+        [Fact]
+        public void CreateFunction_works_when_result_DBNull()
+        {
+            using (var connection = new SqliteConnection("Data Source=:memory:"))
+            {
+                connection.Open();
+                connection.CreateFunction<object>("test", () => DBNull.Value);
+
+                var result = connection.ExecuteScalar<object>("SELECT test();");
+
+                Assert.Equal(DBNull.Value, result);
+            }
+        }
+
+        [Fact]
+        public void CreateFunction_works_when_parameter_null_and_type_nullable()
+        {
+            using (var connection = new SqliteConnection("Data Source=:memory:"))
+            {
+                connection.Open();
+                connection.CreateFunction("test", (long? x) => x == null);
+
+                var result = connection.ExecuteScalar<long>("SELECT test(NULL);");
+
+                Assert.Equal(1L, result);
+            }
+        }
+
+        [Fact]
+        public void CreateFunction_works_when_parameter_null_but_type_not_nullable()
+        {
+            using (var connection = new SqliteConnection("Data Source=:memory:"))
+            {
+                connection.Open();
+                connection.CreateFunction("test", (long x) => x);
+
+                var result = connection.ExecuteScalar<long>("SELECT test(NULL);");
+
+                Assert.Equal(0L, result);
```

Comment:
This behavior is odd (when specifying NULL for non-nullable parameters), but it matches how SQLite will coerce NULL when you ask for a value of a specific type...

Reply:
The other option was to throw. I'll discuss with the team.

Reply:
We decided to throw. I'll make the change.